src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.System/Components/Commands.cs

```diff
@@ -91,7 +96,11 @@ internal static List<Result> GetSystemCommands(bool isUefi, bool splitRecycleBin
                     IcoPath = $"Images\\sleep.{iconTheme}.png",
                     Action = c =>
                     {
-                        return ResultHelper.ExecuteCommand(confirmCommands, Resources.Microsoft_plugin_sys_sleep_confirmation, () => NativeMethods.SetSuspendState(false, true, true));
```

Comment:
For laptops that are still using S1-S3 instead of S0-idle, does the new "monitor power off" message put the device into the correct sleep state? Do we need to branch the actions into 2 cases? For example:

```cs
NativeMethods.GetPwrCapabilities(out SystemPowerCapabilities powerCapabilities);
if (powerCapabilities.AoAc) // modern standby is enabled
{
    return ResultHelper.ExecuteCommand(confirmCommands, Resources.Microsoft_plugin_sys_sleep_confirmation, () => Task.Run(() => _ = NativeMethods.SendMessage(HWNDBROADCAST, WMSYSCOMMAND, SCMONITORPOWER, POWEROFF)));
}
return ResultHelper.ExecuteCommand(confirmCommands, Resources.Microsoft_plugin_sys_sleep_confirmation, () => NativeMethods.SetSuspendState(false, true, true));
```

NativeMethods.cs for GetPwrCapabilities
```cs
[DllImport("Powrprof.dll", SetLastError = true)]
public static extern bool GetPwrCapabilities(out SystemPowerCapabilities lpSystemPowerCapabilities);
```

ref:
* https://learn.microsoft.com/en-us/windows/win32/api/powerbase/nf-powerbase-getpwrcapabilities
* https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-system_power_capabilities
* https://github.com/microsoft/PowerToys/blob/main/src/modules/awake/Awake/Core/Models/SystemPowerCapabilities.cs

Reply:
Hi @der3318,
Thank you for your comment!
Indeed, I could not test the changes on a system with traditional S1-3 power modes. It is also not clear, where the boundaries are (until which Windows version for example) and whether the change makes the sleep function worse on those systems or not. 
I will investigate deeper this topic. Your code suggestion seems to be logical, thank you very much!!

Reply:
Hi @der3318 
I have tested the code you proposed. Albight it seems logical, it did not work for me, the GetPwrCapabilities function did return false response, AoAc was false on my system, which is faulty. I've experimented with it, but it was always the same.
My conclusion in the current situation is that in this form I cannot extend the code.
I will keep analysing the problem. Just letting you know. 
Thanks for the contribution!

Reply:
Thank you for the effort. Sadly I am not aware of any issue regarding API GetPwrCapabilities and the field AoAc, which has been publicly used since at least XP.

Just to confirm 😊 - is it possible that you are not marshaling the struct SYSTEM_POWER_CAPABILITIES correctly? As I just noticed that [MSFT's documentation regarding this struct](https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-system_power_capabilities) is inaccurate. If you take a look at the actual winnt.h (e.g., the one under C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um\), it should be:

```diff
typedef struct {
...
  BOOLEAN                 ProcessorThrottle;
  BYTE                    ProcessorMinThrottle;
+ #if (NTDDI_VERSION < NTDDI_WINXP)
  BYTE                    ProcessorThrottleScale;
  BYTE                    spare2[4];
+ #else
  BYTE                    ProcessorMaxThrottle;
  BOOLEAN                 FastSystemS4;
  BOOLEAN                 Hiberboot;
  BOOLEAN                 WakeAlarmPresent;
  BOOLEAN                 AoAc;
+ #endif
  BOOLEAN                 DiskSpinDown;
...
} SYSTEM_POWER_CAPABILITIES, *PSYSTEM_POWER_CAPABILITIES;
```

Reply:
Hi @der3318,
Yes indeed, there was a glitch in the marshaling. I've rewritten it and it works. I thank you for your colaboration!!